fmralign/alignment_methods.py

```diff
@@ -0,0 +1,196 @@
+        Ux, sx, Vx = linalg.svd(X, full_matrices=0)
+        A = np.diag(sy).dot(Uy.T).dot(Ux).dot(np.diag(sx))
+        U, s, V = linalg.svd(A)
+        R = Vy.T.dot(U).dot(V).dot(Vx)
+
+    if scaling:
+        sc = s.sum() / (np.linalg.norm(X) ** 2)
+    else:
+        sc = 1
+    return R, sc
+
+
+def optimal_permutation(X, Y):
+    """Compute the optmal permutation matrix of X toward Y
+    Parameters
+    ----------
+    X: (n_features, n_samples) nd array
+        source data
+    Y: (n_features, n_samples) nd array
+        target data
+
+    Returns
+    ----------
+    permutation : (n_features, n_features) nd array
+        transformation matrix
+    """
+    dist = pairwise_distances(X, Y)
+    u = linear_assignment(dist)
+    permutation = scipy.sparse.csr_matrix(
+        (np.ones(X.shape[0]), (u[:, 0], u[:, 1]))).T
+    return permutation
+
+
+class Alignment(BaseEstimator, TransformerMixin):
+    def __init__(self):
+        pass
+
+    def fit(self, X, Y):
+        pass
+
+    def transform(self, X):
+        pass
+
+
+class Identity(Alignment):
+    """The simplest kind of alignment to be used as a baseline for benchmarks. RX = X
+    """
+
+    def transform(self, X):
+        """returns X"""
+        return X
+
+
+class ScaledOrthogonalAlignment(Alignment):
+    """Compute a mixing matrix R and a scaling sc such that
+    frobenius norm ||sc RX - Y||^2 is minimized and
+    R is an orthogonal matrix
+
+    Parameters
+    ---------
+    scaling : boolean, optional
+    Determines whether a scaling parameter is applied to improve transform.
+    R : optimal transform
+    """
+
+    def __init__(self, scaling=True):
+        self.scaling = scaling
+        self.scale = None
+
+    def fit(self, X, Y):
+        """ Fit orthogonal R s.t. ||sc RX - Y||^2
+        ----------
+        X: (n_timeframes, n_features) nd array
+            source data
+        Y: (n_timeframes, n_features) nd array
+            target data
+        """
+        R, sc = scaled_procrustes(X, Y, scaling=self.scaling)
+        self.scale = sc
+        self.R = sc * R
+        return self
+
+    def transform(self, X):
+        """Transform X using optimal transform computed during fit.
+        """
+        return self.scale * self.R.dot(X)
+
+
+class RidgeAlignment(Alignment):
+    """ Compute a mixing matrix R such that
+    frobenius norm || XR - Y ||^2 + alpha ||R||^2 is minimized with built-in cross-validation
+
+    Parameters
+    ----------
+    alpha : numpy array of shape [n_alphas]
+        Array of alpha values to try. Regularization strength; must be a positive float. Regularization
+        improves the conditioning of the problem and reduces the variance of
+        the estimates. Larger values specify stronger regularization.
+        Alpha corresponds to ``C^-1`` in other linear models.
+    gcv : int, cross-validation generator or an iterable, optional
+    Determines the cross-validation splitting strategy. Possible inputs for cv are:
```

Comment:
sorry, i wan unclear; This should be called cv.
Please stick to scikit learn notations.